drivers/hmis/app/models/hmis/hud/enrollment.rb

```diff
@@ -403,11 +403,11 @@ def unit_occupied_on(date = Date.current)
 
   def build_synthetic_intake_assessment
     assessment = Hmis::Hud::CustomAssessment.new(
-      enrollment: self,
-      client: client,
-      data_source: data_source,
+      enrollment_id: enrollment_id,
+      personal_id: personal_id,
       assessment_date: entry_date,
-      user: user, # same user that's on the enrollment
```

Comment:
what happened is: if `Enrollment.UserID` is not real, then `user: user` would set `data_source_id: nil`, because of the way the user is related via data source(see `hmis_enrollment_relation`). sneaky!

to avoid all that, I updated each of these to just set the ids. It also helps us avoid database lookups for the whole client and data source records if they're not already on hand.

Reply:
Wow, sneaky indeed. Thank you!